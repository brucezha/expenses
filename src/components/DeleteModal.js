import React from 'react';
import Modal from 'react-modal';
import { Button } from '@material-ui/core';

const DeleteModale = (props) => (
    <Modal
        isOpen={!!props.deleteSelected}
        onRequestClose={props.handleClearDelete}
        ariaHideApp={false}
        closeTimeoutMS={400}
        className="modal"
    >
        <h3 className="modal__title">Are you sure you want to delete this entry?</h3>
        <div className="modal__body">
            <Button onClick={props.remove} variant="contained" style={{ fontSize: '15px' }} color="secondary">Yes</Button>
        &nbsp;
        &nbsp;
        <Button onClick={props.handleClearDelete} variant="contained" style={{ fontSize: '15px' }} color="primary">Actually No</Button>
        </div>
    </Modal>
);
export default DeleteModale;