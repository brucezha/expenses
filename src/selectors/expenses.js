import moment from 'moment';

export default (expenses, { cardName, text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const startDateMatch = startDate ? startDate.isSameOrBefore(moment(expense.createdAt), 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(moment(expense.createdAt), 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        const cardMatch = expense.card.toLowerCase().includes(cardName.toLowerCase());

        return startDateMatch && endDateMatch && textMatch && cardMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};