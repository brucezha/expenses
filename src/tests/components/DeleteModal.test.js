import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import Button from '@material-ui/core/Button';
import Modal from 'react-modal';
import DeleteModal from '../../components/DeleteModal';

let history, wrapper

beforeEach(() => {
    history = { push: jest.fn() };
    wrapper = shallow(
        <DeleteModal />
    );
});

test('should render deleteModal correctly', () => {
    expect(wrapper).toMatchSnapshot();
});