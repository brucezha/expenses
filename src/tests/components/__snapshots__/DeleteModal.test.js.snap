// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render deleteModal correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={400}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Are you sure you want to delete this entry?
  </h3>
  <div
    className="modal__body"
  >
    <WithStyles(ForwardRef(Button))
      color="secondary"
      style={
        Object {
          "fontSize": "15px",
        }
      }
      variant="contained"
    >
      Yes
    </WithStyles(ForwardRef(Button))>
       
    <WithStyles(ForwardRef(Button))
      color="primary"
      style={
        Object {
          "fontSize": "15px",
        }
      }
      variant="contained"
    >
      Actually No
    </WithStyles(ForwardRef(Button))>
  </div>
</Modal>
`;
